src/Jackalope/Version/VersionHandler.php

```diff
@@ -119,27 +129,63 @@ public function checkinItem($path)
 
         // TODO set subgraph to read only
 
-        $versionHistoryNode = $node->getPropertyValue('jcr:versionHistory');
+        /** @var NodeInterface $versionHistoryNode */
+        $versionHistoryUuid = $node->getProperty('jcr:versionHistory')->getString();
+        $versionHistoryNode = $this->objectManager->getNodeByIdentifier($versionHistoryUuid, 'Version\VersionHistory');
 
         // FIXME add some kind of sharding
         // should avoid to have too many nodes on the same level
+        /** @var NodeInterface $versionNode */
```

Comment:
i think this is unneeded, NodeInterface::addNode already says this in the return type